src/com/metacube/get2019/shop.java

```diff
@@ -0,0 +1,250 @@
+package com.metacube.get2019;
+
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Scanner;
+class Item
+{
+int id;
+String itemName;
+double price;
+
+/*
+* This method is used to set values of 3 properties and to print them.
+* @param id This is first parameter to set id of item
+* @param itemName this is second parameter to set itemname
+* @param price this is third parameter to set price of item
+*/
+public Item(int id, String itemName, double price){
+this.id=id;
+this.itemName=itemName;
+this.price=price;
+System.out.println(id+"\t\t"+itemName+"\t\t"+price);
+}
+
+/*
+* This method is used to return value of itemname
+* @return string type itemname
+*/
+public String getItemName(){
+return itemName;
+}
+
+/*
+* This method is used to return value of id.
+* @return id of item
+*/
+public int getid(){
+return id;
+}
+
+/*
+* This method is used to return the price of the item.
+* @return price of item
+*/
+public double getprice(){
+return price;
+}
+}
+
+/*
+ * This class has the attributes pro(Product),quant(Quantity), 
+ * pid(ProductID) and price(Cost) 
+ * This class also has all the functions of a Shopping Cart
+ */
+class ShoppingCart
+{
+	//ArrayList are for storing the input
+	ArrayList<String> pro=new ArrayList<String>();
+	ArrayList<Integer> quant=new ArrayList<Integer>();
+	ArrayList<Integer> pid=new ArrayList<Integer>();
+	ArrayList<Double> price=new ArrayList<Double>();
+	Scanner ab=new Scanner(System.in);
+	/*
+	 * addItem() adds items in the cart 
+	 * while taking the input from the customer
+	 */
+	void addItem(ArrayList<Item> itemList)
+	{
+		System.out.print("enter the item number = ");
+		Scanner ab=new Scanner(System.in);
+		int itemId=ab.nextInt();
+		Iterator<Item> iterator=itemList.iterator();
+		while(iterator.hasNext())
+		{	
+			Item i=(Item)iterator.next();
+			if (i.getid()==itemId)
+			{
+				pro.add(i.getItemName());
+				System.out.print("Enter quantity for the item = ");
+				int itemQuantity=ab.nextInt();
+				quant.add(itemQuantity);
+				price.add(i.getprice());
+				pid.add(i.getid());
+				System.out.println("Item is successfully added into cart");
+			}
+		}
+	}
+	/*
+	 * update() is for changing the quantity of a particular item in the cart
+	 * as required by the customer
+	 */
+	void update()
+	{
+		if(pid.size()==0)
+		{
+			System.out.println("No item in the cart!! Please first enter some!!");
+		}
+		else
+		{
+			System.out.println("Enter the Sid of the product to be updated");

```

Comment:
Do not confuse the user, while adding the item in the card while adding the item on line no 70  you mentioned enter the item number and you are asking end user to provide the SID, try to be stick on the same detail.

Reply:
rectified